### PR TITLE
feat: add bria-client to stablesats

### PR DESCRIPTION
**Pull request by thevaibhav-dixit**
Adds bria to stablesats

### CODE REVIEW

File: bria-client/src/client/bria_client.rs [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+            tonic::metadata::MetadataValue::try_from(key)
+                .map_err(|_| BriaClientError::CouldNotCreateMetadataValue)?,
+        );
+        Ok(request)
+    }
+
+    pub async fn onchain_address(&mut self) -> Result<OnchainAddress, BriaClientError> {
+        match self.get_address().await {
+            Ok(addr) => Ok(addr),
+            Err(_) => self.new_address().await,
```
Comment (bodymindarts) [inline bria-client/src/client/bria_client.rs:51]: We shouldn't eat this error - only do new_address if the  error is NotFound.

File: bria-client/src/client/config.rs [hunk 2]
```diff
@@ -0,0 +1,19 @@
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, Deserialize, Serialize, Default)]
+pub struct BriaClientConfig {
+    #[serde(default = "default_url")]
+    pub url: String,
+    #[serde(default)]
+    pub key: String,
```
Comment (bodymindarts) [inline bria-client/src/client/config.rs:8]: can we be more specific here? `profile_api_key` perhaps?

File: bria-client/src/client/config.rs [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    #[serde(default)]
+    pub wallet_name: String,
+    #[serde(default)]
+    pub payout_queue_name: String,
+    #[serde(default)]
+    pub external_id: String,
```
Comment (bodymindarts) [inline bria-client/src/client/config.rs:14]: How about `onchain_address_external_id`?

File: bria-client/src/client/bria_client.rs [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+    proto_client: ProtoClient,
+}
+
+impl BriaClient {
+    pub async fn connect(config: BriaClientConfig) -> Result<Self, BriaClientError> {
+        let proto_client = ProtoClient::connect(config.url.clone())
+            .await
+            .map_err(|_| BriaClientError::ConnectionError(config.url.clone()))?;
+
+        if config.key.is_empty() {
```
Comment (bodymindarts) [inline bria-client/src/client/bria_client.rs:25]: This is too late to validate. It should be asserted in cli.

File: bria-client/src/error.rs [hunk 5]
```diff
@@ -0,0 +1,25 @@
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum BriaClientError {
+    #[error("Couldn't connect to bria at url: {0}")]
+    ConnectionError(String),
+    #[error("Bria key cannot be empty")]
+    EmptyKey,
```
Comment (bodymindarts) [inline bria-client/src/error.rs:8]: I don't think we should need this error.

File: bria-client/src/error.rs [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    #[error("Couldn't create MetadataValue")]
+    CouldNotCreateMetadataValue,
+    #[error("Couldn't find address for the given external_id")]
+    AddressNotFound,
```
Comment (bodymindarts) [inline bria-client/src/error.rs:12]: When is this returned? This should be handled internally.

File: cli/src/config.rs [hunk 7]
```diff
@@ -62,7 +67,7 @@ impl Config {
         };
 
         config.db.pg_con = stablesats_pg_con;
-
+        config.bria.key = bria_key;
```
Comment (bodymindarts) [inline cli/src/config.rs:70]: Here is where to validate that the key exists.

File: hedging/src/okex/job/adjust_funding.rs [hunk 8]
```diff
@@ -149,10 +151,17 @@ pub(super) async fn execute(
                             &tracing::field::display(String::from(client_id)),
                         );
 
-                        let amount_in_sats = amount * SATS_PER_BTC;
-                        let memo: String = format!("deposit of {amount_in_sats} sats to OKX");
-                        let _ = galoy
-                            .send_onchain_payment(deposit_address, amount_in_sats, Some(memo), 1)
+                        let amount_in_sats = (amount * SATS_PER_BTC)
```
Comment (bodymindarts) [inline hedging/src/okex/job/adjust_funding.rs:154]: This conversion should happen in the client. Accept a decimal in the interface.

File: bria-client/src/error.rs (hunk 6 above)
Comment (thevaibhav-dixit) [reply bria-client/src/error.rs:12]: when we call ```onchain_address()```, it first calls ```get_address()```. suppose we don't have an address for a given ```onchain_address_external_id``` we use ```Err(BriaClientError::AddressNotFound) => self.new_address().await``` to call ```new_address()```

File: bria-client/tests/bria_client.rs [hunk 9]
```diff
@@ -0,0 +1,50 @@
+use std::env;
+
+use ::bria_client::*;
+
+fn client_configuration() -> BriaClientConfig {
+    let url = "http://localhost:2742".to_string();
+    let key = env::var("BRIA_KEY").expect("BRIA_KEY not set");
```
[the remark this answers is not in the document]
Comment (bodymindarts) [reply bria-client/tests/bria_client.rs]: https://github.com/GaloyMoney/bria/blob/3a1ec581ec0d2f0e79f4ecd8b85fd7ef161be769/src/dev_constants.rs#L4

**Review by bodymindarts — APPROVED** [review]

**Pull request merged**